WooCommerce/Classes/ViewRelated/ReusableViews/SwiftUI Components/SegmentedView.swift

```diff
@@ -0,0 +1,63 @@
+import SwiftUI
+
+struct SegmentedView<Content: View>: View {
+    @State private var selectionState: Int {
+        didSet {
+            selection = selectionState
+        }
+    }
+    @Binding private var selection: Int
+    private let views: [Content]
+
+
+    /// Creates a picker that displays a custom label.
+    ///
+    /// - Parameters:
+    ///     - selection: A binding to a property that determines the
+    ///       currently-selected option.
+    ///     - views: A list of view that contains the set of options.
+    public init(selection: Binding<Int>, views: [Content]) {
+        _selectionState = State(initialValue: selection.wrappedValue)
+        _selection = selection
+        self.views = views
+    }
+
+    var body: some View {
+        HStack(spacing: 0) {
+            ForEach(0..<views.count) { (index) in
+                VStack(spacing: 0) {
+                    getContentView(index)
+                    if index == selection {
+                        Rectangle()
+                            .frame(height: 3)
+                            .foregroundColor(Color(.brand))
+                    }
+                    else {
+                        Rectangle()
+                            .frame(height: 3)
+                            .foregroundColor(Color(.clear))
+                    }
+                }
+            }
+        }
+    }
+
+    func getContentView(_ index: Int) -> some View {
+        return views[index]
+            .frame(maxHeight: .infinity)
```

Comment:
I'm curious, why `.infinity` and not something like the device height?

Reply:
That's a good question. The main reason is that since the SegmentedView was implemented with the reuse in mind, and since a developer can add inside anything (not only text), we don't know the height of the content. So there may be cases where:
1) The `SegmentedView` is contained inside a `ScrollView`, so in that case, it's wrong to declare the `maxHeight` equal to the height of the device, maybe the content of the `SegmentedView` is greater than the device height.
2) The content inside can be dynamic, and we don't know the maximum height. Imagine a simple Text that will have different sizes based on the settings of their device.

Reply:
Thanks @pmusolino !